liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -815,6 +799,43 @@ public void undeploy(final Archive<?> archive) throws DeploymentException
          log.exiting(className, "undeploy");
       }
    }
+ 
+   private void deleteWithRetries(File archveToDelete, boolean isFailSafeUndeployment, int retries, int retryInterval) throws DeploymentException {
+      int retryNo = -1;
+      boolean complete = false;
+      do {
+         if (retryNo >= 0) {
+            try {
+               Thread.sleep(retryInterval);
+            } catch (InterruptedException e) {
+               throw new DeploymentException("Interrupted Exception", e);
```

Comment:
Let's have a better exception message here. Maybe "Interrupted while trying to delete <filename>"